scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,191 @@
+import { writeFile, readFile } from 'fs/promises';
+import { GlobSync } from 'glob';
+
+const validMessageTypes = [
+  'Known Issue',
+  'Compatibility Note',
+  'New Functionality',
+  'Improvement',
+  'Fixed Issue',
+  'Updated Dependencies'
+] as const;
+
+interface Message {
+  packageNames: string[];
+  commit: string;
+  type: typeof validMessageTypes[number];
+  version: string;
+  message: string;
+}
+
+function getPackageName(changelog: string): string {
+  const packageNameRegex = /^# @sap-cloud-sdk\/(.+)$/m;
+  const title = changelog.match(packageNameRegex);
+
+  if (!changelog.startsWith('# @sap-cloud-sdk/') || !title || !title[1]) {
+    throw new Error('Changelog should have package name on first line');
+  }
+
+  return title[1];
+}
+
+function splitByVersion(changelog: string): any[] {
```

Comment:
[nth] I guess this extracts the version and content related to it.I would prefer a return type to represent this {version,content} object.